Neos.Diff/Classes/SequenceMatcher.php

```diff
@@ -0,0 +1,723 @@
+     */
+    private $junkCallback = null;
+
+    /**
+     * @var array The first sequence to compare against.
+     */
+    private $a = null;
+
+    /**
+     * @var array The second sequence.
+     */
+    private $b = null;
+
+    /**
+     * @var array Array of characters that are considered junk from the second sequence. Characters are the array key.
+     */
+    private $junkDict = [];
+
+    /**
+     * @var array Array of indices that do not contain junk elements.
+     */
+    private $b2j = [];
+
+    /**
+     * @var array
+     */
+    private $options = [];
+
+    /**
+     * @var array
+     */
+    private $defaultOptions = [
+        'ignoreNewLines' => false,
+        'ignoreWhitespace' => false,
+        'ignoreCase' => false
+    ];
+
+    /**
+     * The constructor. With the sequences being passed, they'll be set for the
+     * sequence matcher and it will perform a basic cleanup & calculate junk
+     * elements.
+     *
+     * @param string|array $a A string or array containing the lines to compare against.
+     * @param string|array $b A string or array containing the lines to compare.
+     * @param string|array $junkCallback Either an array or string that references a callback function (if there is one) to determine 'junk' characters.
+     */
+    public function __construct($a, $b, $junkCallback = null, $options)
+    {
+        $this->a = null;
+        $this->b = null;
+        $this->junkCallback = $junkCallback;
+        $this->setOptions($options);
+        $this->setSequences($a, $b);
+    }
+
+    public function setOptions($options)
+    {
+        $this->options = array_merge($this->defaultOptions, $options);
+    }
+
+    /**
+     * Set the first and second sequences to use with the sequence matcher.
+     *
+     * @param string|array $a A string or array containing the lines to compare against.
+     * @param string|array $b A string or array containing the lines to compare.
+     */
+    public function setSequences($a, $b)
+    {
+        $this->setSeq1($a);
+        $this->setSeq2($b);
+    }
+
+    /**
+     * Set the first sequence ($a) and reset any internal caches to indicate that
+     * when calling the calculation methods, we need to recalculate them.
+     *
+     * @param string|array $a The sequence to set as the first sequence.
+     */
+    public function setSeq1($a)
+    {
+        if (!is_array($a)) {
+            $a = str_split($a);
+        }
+        if ($a == $this->a) {
+            return;
+        }
+
+        $this->a = $a;
+        $this->matchingBlocks = null;
+        $this->opCodes = null;
+    }
+
+    /**
+     * Set the second sequence ($b) and reset any internal caches to indicate that
+     * when calling the calculation methods, we need to recalculate them.
+     *
+     * @param string|array $b The sequence to set as the second sequence.
+     */
```

Comment:
@beautify code.

Oh and everyone reading this: Karsten also has a track record in reviewing minified JavaScript ;-) SCNR